### PR TITLE
Fixes #12427: Use the new class_prefix for package methods

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/12427

### CODE REVIEW

**Comment by amousset** [comment]
Do not merge for now, the classes need to be updated

**Comment by amousset** [comment]
Commit modified

File: tree/30_generic_methods/package_state_options.cf [hunk 1]
```diff
@@ -66,13 +63,12 @@ bundle agent package_state_options(name, version, architecture, provider, state,
                              ifvarclass => "should_report";
       "class copy"             usebundle => _classes_copy("ncf_package_${canonified_state}_${canonified_name}", "${old_class_prefix}"),
                               ifvarclass => "ncf_package_${canonified_state}_${canonified_name}_reached";
-      "new result classes"     usebundle => _classes_copy("${class_prefix}_action", "${class_prefix}"),
-                              ifvarclass => "${class_prefix}_action_reached";
+      "new class copy"         usebundle => _classes_copy("ncf_package_${canonified_state}_${canonified_name}", "${class_prefix}"),
```
Comment (ncharles) [inline tree/30_generic_methods/package_state_options.cf]: it should be ncf_package_${report_param}

File: tree/30_generic_methods/package_install_version_cmp_update.cf [hunk 2]
```diff
@@ -41,8 +41,8 @@ bundle agent package_install_version_cmp_update(package_name, version_comparator
       "canonified_package_name" string => canonify("${package_name}");
       "old_class_prefix"        string => "package_install_${canonified_package_name}";
       "args"                     slist => { "${package_name}", "${version_comparator}", "${package_version}", "${action}", "${update_policy}" };
-      "report_param"      string => join("_", args);
-      "class_prefix"      string => canonify("package_install_${report_param}");
+      "report_param"            string => join("_", args);
```
Comment (ncharles) [inline tree/30_generic_methods/package_install_version_cmp_update.cf]: package_install_version_cmp.cf should also be modified

File: tree/30_generic_methods/package_absent.cf [hunk 3]
```diff
@@ -56,9 +53,8 @@ bundle agent package_absent(name, version, architecture, provider)
                              ifvarclass => "should_report";
       "class copy"             usebundle => _classes_copy("ncf_package_absent_${canonified_name}", "${old_class_prefix}"),
                               ifvarclass => "ncf_package_absent_${canonified_name}_reached";
-      "new result classes"     usebundle => _classes_copy("${class_prefix}_action", "${class_prefix}"),
-                              ifvarclass => "${class_prefix}_action_reached";
+      "new class copy"         usebundle => _classes_copy("ncf_package_absent_${canonified_name}", "${class_prefix}"),
+                              ifvarclass => "ncf_package_absent_${canonified_name}_reached";
```
Comment (ncharles) [inline tree/30_generic_methods/package_absent.cf]: ncf_package_${name}_${version}_${architecture}_${provider}_absent ?

File: tree/30_generic_methods/package_check_installed.cf [hunk 4]
```diff
@@ -31,11 +31,10 @@ bundle agent package_check_installed(package_name)
 {
   vars:
       "old_class_prefix"    string => canonify("package_check_installed_${package_name}");
-      "promisers"          slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
-      "class_prefix"      string => canonify(join("_", "promisers"));
       "args"                 slist => { "${package_name}" };
       "called_class_prefix" string => canonify("package_install_${package_name}");
```
Comment (ncharles) [inline tree/30_generic_methods/package_check_installed.cf]: it should be
package_install_${package_name}___latest_verify_false

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Closing for now.

**Comment by ncharles** [comment]
Commit modified

**Pull request closed without merging**